### PR TITLE
Sanitizers support

### PR DESCRIPTION
**Pull request by tmiasko**
Documentation for existing sanitizers support in rustc, as well as changes proposed in https://github.com/rust-lang/rust/pull/65241.

### CODE REVIEW

File: src/sanitizers.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  Array cookie:            ac
+  Intra object redzone:    bb
+  ASan internal:           fe
+  Left alloca redzone:     ca
+  Right alloca redzone:    cb
+  Shadow gap:              cc
+==10029==ABORTING
+```
+
+Use of uninitialized memory detected by MemorySanitizer. Note that we are using
```
Comment (mark-i-m) [inline src/sanitizers.md]: Perhaps make the title a markdown heading or bolded or something?

Comment (tmiasko) [reply src/sanitizers.md]: Added an introductory paragraph to examples and additional headings.

**Comment by tmiasko** [comment]
I would probably wait for mentioned PR in rustc to get merged first. While
almost everything described here is already true, the recommendation about using
`-Zbuild-std` isn't.

**Comment by tmiasko** [comment]
I extended "How are sanitizers implemented in rustc?" section, and removed everything else.

**Comment by mark-i-m** [comment]
It looks like CI is failing

File: src/sanitizers.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  after return, double free, invalid free, memory leaks.
+* [LeakSanitizer][clang-lsan] a run-time memory leak detector.
+* [MemorySanitizer][clang-msan] a detector of uninitialized reads.
+* [ThreadSanitizer][clang-tsan] a fast data race detector.
+
+## How to use the sanitizers?
+
+To enable a sanitizer compile with `-Zsanitizer=...` option, where value is one
+of `address`, `leak`, `memory` or `thread`. For more details how to use
+sanitizers please refer to rustc book.
```
Comment (nikomatsakis) [inline src/sanitizers.md]: For some reason I can't make suggestions on this repo, but this should be

```
please refer to [the unstable book](https://doc.rust-lang.org/unstable-book/).
```

File: src/sanitizers.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## How are sanitizers implemented in rustc?
+
+The implementation of sanitizers relies entirely on LLVM. It consists of
+compile time instrumentation passes and runtime libraries. The role rustc plays
+in the implementation is limited to the execution of following steps:
```
Comment (nikomatsakis) [inline src/sanitizers.md]: Nit: executon of *the* following steps

File: src/sanitizers.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+1. The sanitizer runtime libraries are part of [compiler-rt] project, and [will
```
Comment (nikomatsakis) [inline src/sanitizers.md]: Nit: *the* compiler-rt project

**Comment by tmiasko** [comment]
Thanks. I applied the suggestions.

**Pull request merged**